packages/snaps-jest/src/matchers.ts

```diff
@@ -179,35 +184,133 @@ export const toSendNotification: MatcherFunction<
   return { message, pass };
 };
 
-export const toRender: MatcherFunction<[expected: Component]> = function (
+/**
+ * Serialise a JSX prop to a string.
+ *
+ * @param prop - The JSX prop.
+ * @returns The serialised JSX prop.
+ */
+function serialiseProp(prop: unknown): string {
+  if (typeof prop === 'string') {
+    return `"${prop}"`;
+  }
+
+  return `{${JSON.stringify(prop)}}`;
+}
+
+/**
+ * Serialise JSX props to a string.
+ *
+ * @param props - The JSX props.
+ * @returns The serialised JSX props.
+ */
+function serialiseProps(props: Record<string, unknown>): string {
+  return Object.entries(props)
+    .filter(([key]) => key !== 'children')
+    .sort(([a], [b]) => a.localeCompare(b))
+    .map(([key, value]) => ` ${key}=${serialiseProp(value)}`)
+    .join('');
+}
+
+/**
+ * Serialise a JSX node to a string.
+ *
+ * @param node - The JSX node.
+ * @param indentation - The indentation level. Defaults to `0`. This should not
+ * be set by the caller, as it is used for recursion.
+ * @returns The serialised JSX node.
+ */
+export function serialiseJsx(node: SnapNode, indentation = 0): string {
+  if (Array.isArray(node)) {
+    return node.map((child) => serialiseJsx(child, indentation)).join('');
+  }
+
+  const indent = '  '.repeat(indentation);
+  if (typeof node === 'string') {
+    return `${indent}${node}\n`;
+  }
+
+  if (!node) {
+    return '';
+  }
+
+  const { type, props } = node;
+  const trailingNewline = indentation > 0 ? '\n' : '';
+
+  if (hasProperty(props, 'children')) {
+    const children = serialiseJsx(props.children as SnapNode, indentation + 1);
+    return `${indent}<${type}${serialiseProps(
+      props,
+    )}>\n${children}${indent}</${type}>${trailingNewline}`;
+  }
+
+  return `${indent}<${type}${serialiseProps(props)} />${trailingNewline}`;
+}
+
+const toRenderLegacy: MatcherFunction<[expected: Component]> = function (
   actual,
   expected,
 ) {
   assertHasInterface(actual, 'toRender');
 
   const { content } = actual;
-  const pass = this.equals(content, expected);
+  const expectedElement = getJsxElementFromComponent(expected);
```

Comment:
It'll be a JSX diff for legacy UIs still right? Probably a breaking change

Reply:
What do you mean?

Reply:
The diff printed will be JSX-only now

Reply:
If the `expected` value is not JSX, it will print the raw object, to sort of match the old behaviour. Otherwise it serialises the JSX and prints that. That seems like the best we can do that isn't converting JSX back to the old format 😅

Reply:
Nevertheless its breaking since `ui.content` changed 🤷‍♂️